hack/builder/Dockerfile

```diff
@@ -1,5 +1,18 @@
-FROM quay.io/centos/centos:stream9
+# ARCH argument for selecting the base image based on the architecture.
+ARG ARCH
+FROM quay.io/centos/centos:stream9 as build_s390x
+#Bazel support for s390x is not available in upstream
+#Copy the pre-built bazel-s390x binary to builder image
+ONBUILD COPY ./bazel /usr/bin/bazel
```

Comment:
I agree with @brianmcarey. However, I would avoid to do the build as part of KubeVirt
Would be possible that you publish a container image containing the bazel binary for s390x?
It doesn't necessary need to be an image, a release somewhere can be also another option, but with an image you don't need an extra repository and you can directly reference it.

Reply:
@vamsikrishna-siddu I'm not sure this copy is correct. Probably the `ONBUILD` suggestion was wrong from my part. However, this small example does the trick:
```dockerfile
ARG stage=B
ARG build_stage_var=STAGE${stage}

FROM alpine as STAGEA
RUN echo "a" > /test

FROM centos as STAGEB
RUN echo "b" > /test

FROM $build_stage_var as BUILD

FROM centos
COPY --from=BUILD /test /
```
You can try with:
+ `$ podman build --build-arg "stage=A" -t test  .` -> it creates the file with a
*  `$ podman build --build-arg "stage=B" -t test  .`  -> it creates the file with b
*  `$ podman build " -t test  .`  -> it creates the file with b as it is the default

Reply:
What I like of this approch  is that if the stage a isn't selected, then alpine isn't pulled and the RUN command in STAGEA isn't run

Reply:
Hi @alicefr What i understand when you said publish a container image containing the bazel binary for s390x is that we will use some alpine base image and put the s390x bazel binary inside it and we publish that image to dockerhub and we use that image in the above dockerfile and copy the bazel binary from it. am i correct here?

Reply:
Actually iam currently working on to build the bazel during the kubevirt build by using this script which will build the bazel for s390x  https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh .

Reply:
> I agree with @brianmcarey. However, I would avoid to do the build as part of KubeVirt Would be possible that you publish a container image containing the bazel binary for s390x? It doesn't necessary need to be an image, a release somewhere can be also another option, but with an image you don't need an extra repository and you can directly reference it here.

Regarding this i will discuss with my team and see what i can do here. Thanks

Reply:
> Actually iam currently working on to build the bazel during the kubevirt build by using this script which will build the bazel for s390x https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh .

I would strongly discourage to build bazel as part of KubeVirt builds. Building bazel takes quite a while, it consumes KubeVirt CI resources and if something goes wrong, then it might slow down or block KubeVirt CI.

I think this needs to be something separate and either publish officially by the bazel community or something well supported by IBM.

Reply:
> Hi @alicefr What i understand when you said publish a container image containing the bazel binary for s390x is that we will use some alpine base image and put the s390x bazel binary inside it and we publish that image to dockerhub and we use that image in the above dockerfile and copy the bazel binary from it. am i correct here?

As I mentioned above, it doesn't necessary need to be a container image. But I think here, it will be easier to reference, but it was only a suggestion.

Again, it doesn't need to be either a container image or alpine. The decision how you publish the bazel binary is up to you. If this is done only for kubeVirt, you can use whatever base image you want, even an empty one.